psa-crypto/src/types/key.rs

```diff
@@ -305,6 +305,37 @@ impl Attributes {
         get_attributes_res?;
         Ok(attributes?)
     }
+
+    /// Sufficient size for a buffer to export the key, if supported
+    #[cfg(feature = "with-mbed-crypto")]
+    pub fn export_key_output_size(self) -> Result<usize> {
+        match self.key_type {
+            Type::RsaKeyPair
+            | Type::RsaPublicKey
+            | Type::EccKeyPair { .. }
+            | Type::EccPublicKey { .. } => Ok(psa_crypto_sys::PSA_EXPORT_KEY_OUTPUT_SIZE(
+                self.key_type.try_into()?,
+                self.bits,
+            )),
+            _ => Err(Error::NotSupported),
+        }
+    }
+
+    /// Sufficient buffer size for a signature using the given key, if the key is supported
+    #[cfg(feature = "with-mbed-crypto")]
+    pub fn sign_output_size(self) -> Result<usize> {
+        match self.key_type {
+            Type::RsaPublicKey
+            | Type::RsaKeyPair
+            | Type::EccPublicKey { .. }
+            | Type::EccKeyPair { .. } => Ok(psa_crypto_sys::PSA_SIGN_OUTPUT_SIZE(
```

Comment:
> If the parameters are a valid combination that is not supported by the implementation, this macro must return either a sensible size or 0.

Do we want to return that 0 or do we want to convert that case into an error? Surely a size of 0 is not helpful to anyone

Reply:
You mean checking the return value and return like `NotSupported` if `0`?

Reply:
Yeah, something like that - but I don't know if that contradicts the spec. Tbh we're returning a `Result` anyway so it's not a big deviation